plugins/fastem_roa_import_export.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 13 December 2023
+
+@author: Nandish Patel
+
+Gives ability to use Import ROAs, Export ROAs tool under Help > Development.
+
+Copyright © 2023 Nandish Patel, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not,
+see http://www.gnu.org/licenses/.
+"""
+
+import csv
+from enum import Enum
+import logging
+from typing import Union
+import wx
+
+from odemis.gui.plugin import Plugin
+
+
+class CSVFieldnames(Enum):
+    PROJECT_NAME = "project_name"
+    ROA_NAME = "roa_name"
+    ROA_COORDINATES_L = "roa_coordinates_l"  # xmin (left)
+    ROA_COORDINATES_T = "roa_coordinates_t"  # ymin (top)
+    ROA_COORDINATES_R = "roa_coordinates_r"  # xmax (right)
+    ROA_COORDINATES_B = "roa_coordinates_b"  # ymax (bottom)
+    ROC_2_NAME = "roc_2_name"
+    ROC_2_COORDINATES_L = "roc_2_coordinates_l"  # xmin (left)
+    ROC_2_COORDINATES_T = "roc_2_coordinates_t"  # ymin (top)
+    ROC_2_COORDINATES_R = "roc_2_coordinates_r"  # xmax (right)
+    ROC_2_COORDINATES_B = "roc_2_coordinates_b"  # ymax (bottom)
+    ROC_3_NAME = "roc_3_name"
+    ROC_3_COORDINATES_L = "roc_3_coordinates_l"  # xmin (left)
+    ROC_3_COORDINATES_T = "roc_3_coordinates_t"  # ymin (top)
+    ROC_3_COORDINATES_R = "roc_3_coordinates_r"  # xmax (right)
+    ROC_3_COORDINATES_B = "roc_3_coordinates_b"  # ymax (bottom)
+
+    @staticmethod
+    def list():
+        return list(map(lambda c: c.value, CSVFieldnames))
+
+
+def get_csv_filepath_from_filedialog(parent, message: str, style: int) -> Union[str, None]:
```

Comment:
Why do you make it a separate function? I think it'd feel more logical/connected if it was just another method of the plugin. It could then use `self._parent` instead of receiving `parent` explicitly.